modules/sfp_sociallinks.py

```diff
@@ -0,0 +1,254 @@
+# -------------------------------------------------------------------------------
+
+import json
+
+from spiderfoot import SpiderFootEvent, SpiderFootPlugin
+
+
+class sfp_sociallinks(SpiderFootPlugin):
+
+    meta = {
+        'name': "Social Links",
+        'summary': "Queries mtg-bi.com to gather intelligence from social media platforms and dark web",
+        'flags': ["apikey"],
+        'useCases': ["Footprint", "Investigate", "Passive"],
+        'categories': ["Real World"],
+        'dataSource': {
+            'website': "https://mtg-bi.com/",
+            'model': "COMMERCIAL_ONLY",
+            'references': [
+                "https://docs.osint.rest/"
+            ],
+            'favIcon': "view-source:https://static.tildacdn.com/tild6563-6633-4533-b362-663333656461/favicon.ico",
+            'logo': "https://static.tildacdn.com/tild3935-6136-4330-b561-643034663032/LogoSL.svg",
+            'description': "Social Links provides instruments for OSINT methods "
+            "that are used by the world's leading investigation and law enforcement agencies",
+        }
+    }
+
+    # Default options
+    opts = {
+        'api_key': '',
+    }
+
+    # Option descriptions
+    optdescs = {
+        'api_key': "mtg-bi.com API Key",
+    }
+
+    results = None
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return [
+            "USERNAME",
+            "EMAILADDR",
+            "PHONE_NUMBER"
+        ]
+
+    # What events this module produces
+    # This is to support the end user in selecting modules based on events
+    # produced.
+    def producedEvents(self):
+        return [
+            "GEOINFO",
+            "SOCIAL_MEDIA",
+            "HUMAN_NAME",
+            "JOB_TITLE",
+            "COMPANY_NAME",
+            "PHONE_NUMBER",
+            "ACCOUNT_EXTERNAL_OWNED",
+            "RAW_RIR_DATA"
+        ]
+
+    def query(self, queryString):
+        headers = {
+            'Accept': "application/json",
+            'Authorization': self.opts['api_key']
+        }
+
+        res = self.sf.fetchUrl(
+            queryString,
+            headers=headers,
+            timeout=15,
+            useragent=self.opts['_useragent']
+        )
+
+        if res['content'] is None:
+            return None
+        return json.loads(res['content'])
+
+    def queryTelegram(self, qry, eventName):
+        if eventName == "PHONE_NUMBER":
+            queryString = f"https://osint.rest/api/telegram/user_by_phone?query={qry}"
+        elif eventName == "USERNAME":
+            queryString = f"https://osint.rest/api/telegram/user_by_alias?query={qry}"
+
+        return self.query(queryString)
+
+    def queryFlickr(self, qry, eventName):
+        if eventName == "EMAILADDR":
```

Comment:
What is the purpose of these various query* calls which have a single `if` statement checking the `eventName`? If the function only takes one kind of event type, shouldn't the caller ensure that's what it's called with?